### ISSUE TITLE
Deno Typescript error checking behaves differently from tsc, despite same tsconfig.json

### ISSUE THREAD

**Issue by ericvergnaud** [denoland/deno#21655]
Version: Deno 1.39.0 (v8 12.0.267.8 typescript 5.3.3)

I'm attempting to migrate a project from Node, and getting hundreds of errors.
The project has a tsconfig.json.

`tsc --version` says 5.3.3 (same as deno typescript version)

When I simply run `tsc` on the original project, I don't get any error.
I adjusted file suffixes and converted usage of npm packages to std ones.
When I run `deno test someTest.ts`, I het hundreds of type errors...

BUG:
even in presence of a tsconfig.json, deno does not behave the same as tsc 

Is there a workaround ? How do I get deno to behave the same as tsc ?

**Comment by bartlomieju** [comment]
Do you get the same result if you explicitly specify `--config tsconfig.json`? Deno doesn't pick up this file automatically.

**Comment by ericvergnaud** [comment]
`deno test --config tsconfig.json someTest.ts` triggers the same errors

**Comment by ericvergnaud** [comment]
actually I get `error: unexpected argument '--config' found`

**Comment by bartlomieju** [comment]
Is this repository public? If so could you link to it?

**Comment by ericvergnaud** [comment]
ah, I'll try that

**Comment by ericvergnaud** [comment]
mmm... that doesn't fix the issue, still getting hundreds of type errors

**Comment by ericvergnaud** [comment]
{
  "compilerOptions": {
    "noImplicitAny": true,
    "target": "es2017",
    "module": "esnext",
    "moduleResolution": "node",
    "declaration": true,
    "declarationDir": "dist",
    "sourceMap": true,
    "esModuleInterop": true
  }
}

**Comment by ericvergnaud** [comment]
Here's the repo: https://github.com/compose-lang/compiler/tree/use-deno

**Comment by bartlomieju** [comment]
@ericvergnaud could you provide the output you get in a gist? I'll try running your repo locally, but from I can tell is that some of the options you specify are not supported by Deno (eg. "moduleResolution" or "sourceMap").

CC @dsherret could you take a look here and chime in what's the expected outcome?

**Comment by ericvergnaud** [comment]
I looked at https://docs.deno.com/runtime/manual/advanced/typescript/configuration
setting strict to false, massively reduces the damage

**Comment by ericvergnaud** [comment]
I'm left with one weird error, with the below deno.json: 
{
  "compilerOptions": {
    "esModuleInterop": true,
    "module": "esnext",
    "moduleDetection": "force",
    "strict": false,
    "target": "esnext",
    "declaration": true,
    "declarationDir": "dist",
    "sourceMap": true

  }
}
error: TS5069 [ERROR]: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.
Any clue ?

**Comment by dsherret** [comment]
> setting strict to false, massively reduces the damage

Yeah, Deno uses strict by default, which differs from TypeScript's defaults (TypeScript recommends turning on strict, but it doesn't have it on by default for backwards compatibility reasons).

> error: TS5069 [ERROR]: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.

I think probably deno shouldn't surface errors for declarationDir as it doesn't make sense to use in Deno. You can temporarily fix that error by removing `"declaration": true` and `"declarationDir": "dist"`.

**Comment by bartlomieju** [comment]
Gonna close this one due to lack of activity. As mentioned by David, the difference output is expected because Deno uses strict settings by default. Feel free to open a new issue if you run into similar problem. thanks!
